src/core/server/plugins/plugins_service.ts

```diff
@@ -257,17 +264,28 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
     pluginName: PluginName,
     pluginEnableStatuses: Map<PluginName, { plugin: PluginWrapper; isEnabled: boolean }>,
     parents: PluginName[] = []
-  ): boolean {
+  ): { enabled: boolean; missingDependencies: string[] } {
```

Comment:
I prefer to change this return type to:
```ts
{ enabled: true } | { enabled: false; missingDependencies: string[] }
```

This helps us avoid bugs where we have a state that should be impossible, like:
```ts
{ enabled: true, missingDependencies: ['some-dep'] }
```

Reply:
5d1b4e5